build_modules/lib/src/meta_module.dart

```diff
@@ -262,8 +262,8 @@ Future<List<Module>> _computeModules(
   for (var asset in assets) {
     var content = await reader.readAsString(asset);
     // Skip errors here, dartdevc gives nicer messages.
-    var parsed = parseCompilationUnit(content,
-        name: asset.path, parseFunctionBodies: false, suppressErrors: true);
+    var parsed =
+        parseDirectives(content, name: asset.path, suppressErrors: true);
```

Comment:
Note that if `public` is false then we do actually need the whole compilation unit because `_isEntrypoint` checks for a `main` function. This could change to conditionally parse the entire compilation unit though only if `public` is `true`.

Reply:
Ok, got you. Fixed that. Applying parseDirectives on lib/** files only still gives great memory win on big project :-)